core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1320,6 +1331,175 @@ protected SqlNode performUnconditionalRewrites(
     return node;
   }
 
+  private SqlNode performHiveUdtfRewrite(SqlNode node) {
+    Map<SqlSelect, UdtfInfo> selectUdtfInfo = new HashMap<>();
+     SqlNode rewirte = performHiveUdtfRewriteInternal(node, selectUdtfInfo);
+   //  System.out.println("before:\n" + node.toString() +" \nafter:\n" + rewirte.toString());
+    return node;
+  }
+
+  private SqlNode performHiveUdtfRewriteInternal(SqlNode current,
+                                                 Map<SqlSelect, UdtfInfo> selectUdtfInfos) {
+    System.out.println("start rewrite:" + current);
+
```

Comment:
We don't print anything in general.

Reply:
Sorry for push this test code,remove now!